common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -146,25 +162,51 @@ public float getRenderedHeight() {
         return this.size.getRenderedHeight();
     }
 
+    public OverlaySize getSize() {
+        return size;
+    }
+
+    public OverlayPosition getPosition() {
+        return position;
+    }
+
+    public void setPosition(OverlayPosition position) {
+        this.position = position;
+    }
+
+    public void setVerticalAlignmentOverride(VerticalAlignment verticalAlignmentOverride) {
+        this.verticalAlignmentOverride = verticalAlignmentOverride;
+    }
+
+    public void setHorizontalAlignmentOverride(HorizontalAlignment horizontalAlignmentOverride) {
+        this.horizontalAlignmentOverride = horizontalAlignmentOverride;
+    }
+
     // Return the X where the overlay should be rendered
-    public int getRenderX() {
-        final SectionCoordinates section = OverlayManager.getSection(this.position.getAnchorSection());
-        return switch (this.position.getHorizontalAlignment()) {
-            case Left -> section.x1() + this.position.getHorizontalOffset();
-            case Center -> (int) (section.x1() + section.x2() - this.getWidth()) / 2
-                    + this.position.getHorizontalOffset();
-            case Right -> (int) (section.x2() + this.position.getHorizontalOffset() - this.getWidth());
-        };
+    public float getRenderX() {
+        return getRenderX(this.position, this);
     }
 
     // Return the Y where the overlay should be rendered
-    public int getRenderY() {
-        final SectionCoordinates section = OverlayManager.getSection(this.position.getAnchorSection());
-        return switch (this.position.getVerticalAlignment()) {
-            case Top -> section.y1() + this.position.getVerticalOffset();
-            case Middle -> (int) (section.y1() + section.y2() - this.getHeight()) / 2
-                    + this.position.getVerticalOffset();
-            case Bottom -> (int) (section.y2() + this.position.getVerticalOffset() - this.getHeight());
+    public float getRenderY() {
+        return getRenderY(this.position, this);
+    }
+
+    public static float getRenderX(OverlayPosition position, Overlay overlay) {
```

Comment:
I'm not sure why you made this static and take in an overlay. I think it can (and should) be an instance method. The one place you call it as a static method can just as well be called as an instance method on the given overlay.

(It is also a bit weird to have the same name to a static and instance method)

Reply:
The static method has it is use case and cannot be replaced by the instance method, but I can rename it for clarity.

Reply:
Actually, I cannot come up with a better name, but the use case is in OverlayPosition line 124-125.

Reply:
Why not:

```
        float renderX = overlay.getRenderX(newOverlayPositionTemp);
        float renderY = overlay.getRenderY(newOverlayPositionTemp);
```

or am I missing something?

Reply:
Oh my bad. Fixed.